generator/src/test/java/com/scottlogic/deg/generator/cucumber/steps/DateValueStep.java

```diff
@@ -1,15 +1,22 @@
 package com.scottlogic.deg.generator.cucumber.steps;
 
+import com.scottlogic.deg.generator.cucumber.utils.CucumberTestHelper;
 import com.scottlogic.deg.generator.cucumber.utils.CucumberTestState;
+import com.scottlogic.deg.generator.cucumber.utils.GeneratorTestUtilities;
+import cucumber.api.java.en.Then;
 import cucumber.api.java.en.When;
 
+import java.time.LocalDateTime;
+
 public class DateValueStep {
 
-    public static final String DATE_REGEX = "((\\d{4})-(\\d{2})-(\\d{2}T(\\d{2}:\\d{2}:\\d{2}\\.\\d{3})))$";
+    public static final String DATE_REGEX = "((\\d{4})-(\\d{2})-(\\d{2}T(\\d{2}:\\d{2}:\\d{2}\\.\\d{3})))";
```

Comment:
This permits a `{date}` anywhere in the step definition, rather than requiring it to be at the end.